### PR TITLE
improve few things around build and adds changelog

### PR DESCRIPTION
**Pull request by epinault**
- update deps
- fix license
- add changelog
- improve build checks and matrix

### CODE REVIEW

**Comment by epinault** [comment]
@jtomaszewski if you have time to check this

File: lib/ecto_cursor_based_stream.ex [hunk 1]
```diff
@@ -72,6 +73,7 @@ defmodule EctoCursorBasedStream do
   end
 
   @doc false
+  @spec call(Ecto.Repo.t(), Ecto.Queryable.t(), cursor_based_stream_opts) :: Enumerable.t()
```
Comment (vereis) [inline lib/ecto_cursor_based_stream.ex:76]:
```suggestion
  @spec call(Ecto.Repo.t(), Ecto.Queryable.t(), opts :: Keyword.t()) :: Enumerable.t()
```

File: LICENSE [hunk 2]
```diff
@@ -186,7 +186,7 @@
       same "printed page" as the copyright notice for easier
       identification within third-party archives.
 
-   Copyright [yyyy] [name of copyright owner]
+   Copyright [2023] [Jack Tomaszewski]
```
Comment (vereis) [inline LICENSE:189]: 🤣

File: lib/ecto_cursor_based_stream.ex (hunk 1 above)
Comment (epinault) [reply lib/ecto_cursor_based_stream.ex:76]: mine is more correct, it provide the list of possible. Just need to use union type if you want more possible options

Comment (vereis) [reply lib/ecto_cursor_based_stream.ex:76]: oh sorry I didn't realize it was a type! my bad~

**Comment by epinault** [comment]
thank you @alukasz for the help!

**Pull request merged**